src/orchestration/KubernetesClient.js

```diff
@@ -40,6 +40,26 @@ class KubernetesClient {
    * @returns {Promise<EngineContainerSpec[]>} A promise to a list of engine container specs.
    */
   static async listEngines() {
+    const replicaMap = new Map();
+    try {
+      const replicaSets = await kubeHttpGet('/apis/apps/v1/replicasets');
+      replicaSets.items.forEach((item) => {
+        replicaMap.set(item.metadata.uid, item);
+      });
+    } catch (error) {
+      // Do nothing.
```

Comment:
We should probably log something if this fails, or?

Reply:
I tried logging if it fails, since this will be the same on every "round" it gets very spammy.